### PR TITLE
Add customLogFiles sauce option

### PR DESCRIPTION
**Pull request by ben-stevens-sauce**
### Description
Add "customLogFiles" sauce:option, for uploading extra files at the end of a test.

### Motivation and Context
At the end of a test, some mobile applications will write log files or crash reports to their own internal directory, which can be useful in debugging test failures. We should expose these files where appropriate to aid in problem identification. 

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist
- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/dev/test-configuration-options.md [hunk 1]
```diff
@@ -810,6 +810,37 @@ capabilities.setCapability("sauce:options", sauceOptions);
 
 ---
 
+### `customLogFiles`
+<p><small>| OPTIONAL | LIST | <span className="sauceDBlue">Virtual Devices Only</span> |</small></p>
+
+A list of search filters to enumerate after an app test to locate text files to upload as logs. Files are uploaded with the `.log` extension appended. The search paths are rooted at the application under test:
```
Comment (wswebcreation) [inline docs/dev/test-configuration-options.md:816]: Can we make this a bit more clear by also explaining why people can use it. For example

> If you app creates extra log then you can use the `customLogFiles` to store those extra logs in the "Logs"-tab of the executed automated session. .....

**Review by wswebcreation — CHANGES_REQUESTED** [review]
Hi @ben-stevens-sauce 

Thanks for the PR, I have one small comment and it might be good to add a screenshot where people can find the logs. Maybe with some steps like:

> Go to the executed session in the Sauce Labs dashboard by going to:
> - "Automated > Test Results"
> - search for your session and click on it
> - Go to the "Logs" tab above the video

Then you can create a screenshot where you highlight the steps.

File: docs/dev/test-configuration-options.md [hunk 2]
```diff
@@ -810,6 +810,37 @@ capabilities.setCapability("sauce:options", sauceOptions);
...
+
+Supplying the list `["files/*_log", "*crash*"]` to an Android app test of the package `com.saucelabs.exampleapp` will upload all the files found after the test, that match either of the glob expressions:
+  - `/data/data/com.saucelabs.exampleapp/files/*.log`
+  - `/data/data/com.saucelabs.exampleapp/*crash*`
+
+Supplying the list `["files/*_log", "*crash*"]` to an iOS app test will upload all the files found after the test, that match either of the glob expressions (SIMULATED_DEVICE_PATH and APPLICATION_ID filled in with example values):
+  - `~/Library/Developer/CoreSimulator/Devices/8BF8C5E3-E992-424F-A491-5C673761737C/data/Containers/Data/Application/DBF4A728-9414-4431-9A56-41EC1CBFFA0B/files/*.log`
+  - `~/Library/Developer/CoreSimulator/Devices/8BF8C5E3-E992-424F-A491-5C673761737C/data/Containers/Data/Application/DBF4A728-9414-4431-9A56-41EC1CBFFA0B/*crash*`
+
+In both sets of examples, it is worth nothing that an asterisk can match directory elements as well as characters, so `files/*log` will match both `files/debug.log` and `files/subdir/other.log`
```
Comment (aoleszyn) [inline docs/dev/test-configuration-options.md:830]:
```suggestion
 * Android (path on the emulated device): `/data/data/*PACKAGE_ID*/...`
 * iOS: `*SIMULATED_DEVICE_PATH*/data/Containers/Data/Application/*APPLICATION_ID*/...`. On a macOS filesystem, an example of SIMULATED_DEVICE_PATH would be `~/Library/Developer/CoreSimulator/Devices/*DEVICE_ID*`

The following examples outline how this is handled for the different device types.

Supplying the list `["files/*_log", "*crash*"]` to an Android app test of the package `com.saucelabs.exampleapp` will upload all the files found after the test, that match either of the glob expressions:
  * `/data/data/com.saucelabs.exampleapp/files/*.log`
  * `/data/data/com.saucelabs.exampleapp/*crash*`

Supplying the list `["files/*_log", "*crash*"]` to an iOS app test will upload all the files found after the test, that match either of the glob expressions (SIMULATED_DEVICE_PATH and APPLICATION_ID filled in with example values):
  * `~/Library/Developer/CoreSimulator/Devices/8BF8C5E3-E992-424F-A491-5C673761737C/data/Containers/Data/Application/DBF4A728-9414-4431-9A56-41EC1CBFFA0B/files/*.log`
  * `~/Library/Developer/CoreSimulator/Devices/8BF8C5E3-E992-424F-A491-5C673761737C/data/Containers/Data/Application/DBF4A728-9414-4431-9A56-41EC1CBFFA0B/*crash*`

In both sets of examples, it is worth noting that an asterisk can match directory elements as well as characters, so `files/*log` will match both `files/debug.log` and `files/subdir/other.log`
```

**Review by aoleszyn — APPROVED** [review]

**Review by wswebcreation — APPROVED** [review]

File: docs/dev/test-configuration-options.md [hunk 3]
```diff
@@ -832,6 +834,47 @@ capabilities.setCapability("sauce:options", sauceOptions);
 
 ---
 
+### `customLogFiles`
+<p><small>| OPTIONAL | LIST | <span className="sauceDBlue">Virtual Devices Only</span> |</small></p>
+
+A list of search filters to enumerate after an app test to locate text files to upload as logs. Files are uploaded with the `.log` extension appended. The search paths are rooted at the application under test:
```
Comment (wswebcreation) [inline docs/dev/test-configuration-options.md:840]:
```suggestion
If your app creates extra log then you can use the `customLogFiles` to store those extra logs in the "Logs"-tab of the executed automated session. It needs to be list of search filters that will enumerate after an app test to locate text files to upload as logs. Files are uploaded with the `.log` extension appended. The search paths are rooted at the application under test:
```

File: docs/dev/test-configuration-options.md [hunk 4]
```diff
@@ -832,6 +834,47 @@ capabilities.setCapability("sauce:options", sauceOptions);
 
 ---
 
+### `customLogFiles`
+<p><small>| OPTIONAL | LIST | <span className="sauceDBlue">Virtual Devices Only</span> |</small></p>
+
+If your app creates extra log then you can use the `customLogFiles` to store those extra logs in the "Logs"-tab of the executed automated session. It needs to be list of search filters that will enumerate after an app test to locate text files to upload as logs. Files are uploaded with the `.log` extension appended. The search paths are rooted at the application under test:
```
Comment (aoleszyn) [inline docs/dev/test-configuration-options.md:840]:
```suggestion
If your app creates an extra log then you can use the `customLogFiles` to store those additional logs in the "Logs" tab of the executed automated session. It is created in the form of a list of search filters that enumerate after an app test to locate text files to upload as logs. Files are uploaded with the `.log` extension appended. The search paths are rooted at the application under test:
```

**Pull request merged**